src/audio/pipeline.c

```diff
@@ -376,11 +376,62 @@ static int pipeline_comp_hw_params(struct comp_dev *current,
 	return 0;
 }
 
+static int pipeline_comp_params_neg(struct comp_dev *current,
+				    struct comp_buffer *calling_buf,
+				    struct pipeline_walk_context *ctx,
+				    int dir)
+{
+	struct pipeline_data *ppl_data = ctx->comp_data;
+	uint32_t flags = 0;
+	int err = 0;
+
+	pipe_dbg(ppl_data->p, "pipeline_comp_params_neg(), current->comp.id = %u, dir = %u",
+		 dev_comp_id(current), dir);
+
+	/* return if current is already configured */
+	if (current->state != COMP_STATE_INIT &&
+	    current->state != COMP_STATE_READY) {
+		if (buffer_params_match(calling_buf,
+					&ppl_data->params->params))
+			return 0;
+		/*
+		 * the param is conflict with an active pipeline,
+		 * drop an error and reject the .params() command.
+		 */
+		pipe_err(ppl_data->p, "pipeline_comp_params_neg(): params conflict with existed active pipeline!");
```

Comment:
Will this work is a pcm converter is joining the two pipelines/components ?

Reply:
No, we need to change/remove this when pcm converter added, but since I can't imagine how that work, so returning error as this basic solution, and leaving the advanced solution to modify this as incremental change (or kconfig selection added).

Reply:
ok, you need to use the "walked" status for every buffer (like DAPM with path->walked).

Reply:
> ok, you need to use the "walked" status for every buffer (like DAPM with path->walked).

It is **"walking"** for DAPM path, see here:
```
/* dapm audio path between two widgets */
struct snd_soc_dapm_path {
	const char *name;
...
	u32 walking:1;  /* path is in the process of being walked */
...
};
```